src/schemathesis/specs/openapi/parameters.py

```diff
@@ -0,0 +1,348 @@
+from typing import Any, ClassVar, Dict, List, Tuple
+
+import attr
+
+from ...parameters import Parameter
+from .converter import to_json_schema_recursive
+
+
+@attr.s(slots=True)
+class OpenAPIParameter(Parameter):
+    """A single Open API operation parameter."""
+
+    example_field: ClassVar[str]
+    examples_field: ClassVar[str]
+    nullable_field: ClassVar[str]
+    supported_jsonschema_keywords: ClassVar[Tuple[str, ...]]
+
+    @property
+    def example(self) -> Any:
+        """Iterate over all examples defined for the parameter."""
+        if self._example:
+            return self._example
+        if self._schema_example:
+            # It is processed only if there is no `example` / `examples` in the root, overridden otherwise
+            # https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#fixed-fields-10
+            return self._schema_example
+
+    @property
+    def location(self) -> str:
+        """Where this parameter is located.
+
+        E.g. "query".
+        """
+        return {"formData": "body"}.get(self.raw_location, self.raw_location)
+
+    @property
+    def raw_location(self) -> str:
+        """Open API specific location name."""
+        return self.definition["in"]
+
+    @property
+    def name(self) -> str:
+        """Parameter name."""
+        return self.definition["name"]
+
+    @property
+    def is_required(self) -> bool:
+        return self.definition.get("required", False)
+
+    @property
+    def is_header(self) -> bool:
+        raise NotImplementedError
+
+    @property
+    def _example(self) -> Any:
+        """A not-named example, defined in the parameter root.
+
+        {
+            "in": "query",
+            "name": "key",
+            "type": "string"
+            "example": "foo",   # This one
+        }
+        """
+        return self.definition.get(self.example_field)
+
+    @property
+    def named_examples(self) -> Dict[str, Any]:
```

Comment:
This property is not used now - but need to re-check if the named examples are properly used.